lib/extendOrdWithCustom.js

```diff
@@ -28,12 +28,12 @@ function patchGeneratedOrdResources(destinationObj, sourceObj) {
     return cleanNullProperties(Object.values(destObj));
 }
 
-function compareAndHandleCustomORDContentWithExistingContent(ordContent, customORDContent, logger) {
+function compareAndHandleCustomORDContentWithExistingContent(ordContent, customORDContent) {
     const clonedOrdContent = structuredClone(ordContent);
     for (const key in customORDContent) {
         const propertyType = typeof customORDContent[key];
         if (propertyType !== 'object' && propertyType !== 'array') {
-            logger.warn('Found ord top level primitive ord property in customOrdFile:', key, ', please define it in .cdsrc.json.');
+            log('ord-plugin').warn('Found ord top level primitive ord property in customOrdFile:', key, ', please define it in .cdsrc.json.');
```

Comment:
@zongqichen : Explanation given above(no need to propagate input parameter `logger`).

Reply:
From my opinion, the central logger object is needed. The reasons are:
- if we want to change logger name: 'ord-plugin' -> 'ord', we have to maintain it in all different places
- if we have special configuration to logger, e.g set level, env, or save debug msg into file, central logger can achieve it
- I guess put logger as argument won't spend extra costs? 

And I have an issue wants to put central logger to all necessary places. I learnt it from previous golang project, which we define the logger object at the begging and spread it to all places. I'm not sure how should javascript best practice look like, what's your idea?

Reply:
Yes, refactoring logging should be separate PR. 

If it's a factory we get, I assume we should have a singleton instance created for the entire plugin that we then re-use in different files?

Reply:
Let's do that in a separate PR.